### PR TITLE
chore(ICP-Rosetta): move block range test

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR introduces the following changes:

1. Remove the legacy test for the block range
2. Add a new test using pocket ic for fetching block range

### CODE REVIEW

File: rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs [hunk 1]
```diff
@@ -135,3 +141,133 @@ fn test_fetching_blocks() {
...
+                            method_name: "query_block_range".to_owned(),
+                            parameters: ObjectMap::try_from(query_blocks_request.clone()).unwrap(),
+                        })
+                        .await
+                        .unwrap()
+                        .result
+                        .try_into()
+                        .unwrap();
+
+                    assert_eq!(query_blocks_response.blocks.len(), chain_length as usize);
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs:267]: Would it also make sense to test with a `highest_block_index` less than the total amount of blocks? And a value of `0` for both the `highest_block_index` and `number_of_blocks`? Perhaps these are already sufficiently tested in the unit/integration tests, in which case feel free to ignore this comment!

**Review by mbjorkqvist — APPROVED** [review]
Thanks for this cleanup, @NikolasHai !

File: rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs (hunk 1 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs:267]: I'll do that.

**Review by fsodre — APPROVED** [review]

File: rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs (hunk 1 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/blocks.rs:267]: Done

**Pull request merged**